autoload/async/job.vim

```diff
@@ -190,17 +190,23 @@ function! s:job_stop(jobid) abort
     endif
 endfunction
 
-function! s:job_send(jobid, data) abort
+function! s:job_send(jobid, data, close_stdin) abort
     let l:jobinfo = s:jobs[a:jobid]
     if l:jobinfo.type == s:job_type_nvimjob
         call jobsend(a:jobid, a:data)
+        if a:close_stdin
+          call chanclose(a:jobid, 'stdin')
+        endif
     elseif l:jobinfo.type == s:job_type_vimjob
         if has('patch-8.1.0818')
             call ch_sendraw(l:jobinfo.channel, a:data)
         else
             let l:jobinfo.buffer .= a:data
             call s:flush_vim_sendraw(a:jobid, v:null)
         endif
+        if a:close_stdin
+          call ch_close_in(l:jobinfo.channel)
```

Comment:
This might fail in case the transmit buffer was not empty.

I changed this to the following, but I am not familiar with `sleep`, so I don't know if it's blocking the main thread or not -- it does not seem like it, but I will keep digging and testing

```vimscript
        if a:close_stdin
          while len(l:jobinfo.buffer) != 0
              sleep 1m
          endwhile
          call ch_close_in(l:jobinfo.channel)
        endif
```